service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java

```diff
@@ -76,14 +88,32 @@ protected StepResult handleResourceDeleteException(Exception e, FlightContext co
     // When retrieving shared resources from the landing zone,
     // LZS can encounter the same management exceptions as above
     // To detect this, we need to examine the message for those error codes
-    if (e instanceof LandingZoneServiceApiException && e.getCause() instanceof ApiException) {
-      var msg = e.getCause().getMessage();
-      if (msg != null
-          && missingResourceManagementCodes.stream().anyMatch(code -> msg.contains(code))) {
+    if (e instanceof LandingZoneServiceApiException && e.getCause() instanceof ApiException apiEx) {
+      var code =
+          Optional.ofNullable(apiEx.getMessage())
+              .flatMap(
+                  msg ->
+                      missingResourceManagementCodes.stream()
+                          .filter(c -> msg.contains(c))
+                          .findFirst());
+      if (code.isPresent()) {
+        logger.debug(
+            "LZS encountered management exception {} when deleting resource {} from workspace {}",
```

Comment:
Nit: note in the log msg that we are explicitly ignoring the exception and moving on.